README.md

```diff
@@ -12,103 +12,88 @@ Stage: 1
-import must be asynchronous, otherwise we would break run-to-completion semantics. Dynamic import is
-already a powerful tool in use on many applications.
+The API can take a couple of forms, here are some suggestions:
 
-If we consider **2) Before Parse**, we see that it is ill-advised to laziness is at this point, due to the note at 2.i. Without parsing, we will be unable to have a complete module graph. We want the module graph to be present and complete, so that other work can also be
-done. So, **we do not want to defer parsing**. However, many engines may implement an optimization
-to only partially parse the module, and find early errors and imports.
+```js
+// using import attributes (`lazyInit` to illustrated, but it could also be `deferEval`, say)
+import lazyModuleNamespace from "y" with { lazyInit: true }
 
-If we consider **3) Before Evaluate**, we would relinquish some significant performance advantages.
-However, if those are important, a developer would be able to fall back on dynamic import, and trade
-in sync code for more performance. Secondly, we would change the evaluation order. Presently, we
-have an invariant that all child modules of a parent module will complete evaluation before their
-parent module completes evaluation. What lazy-eval would introduce is a new concept. We would have a
-subgraph that does not evaluate with the parent, but instead waits untill it is called.
+// with an explicit "*" to indiciate the namespace form
+import * as lazyModuleNamespace from "y" with { lazyInit: true }
 
-### Impact on execution order
+// or with a custom keyword:
+lazy import * as yNs from "y";
+```
 
-|                                | Static Import        | Dynamic Import               | "Lazy" Import  |
-|--------------------------------|----------------------|------------------------------|----------------|
-| Top level exectution of module | Load, Parse, Execute |                              |  Load, Parse   |
-| runtime import                 | --                   | Load, Parse, Execute         |  --            |
-| First use                      |                      |                              |  Execute       |
+There are other proposals, but the discussion of the naming may derail the discussion of the
+semantics, which should be agreed on first and would likely guide the design of the API.
+
+## Semantics
 
-The proposal in it's simplest form will pause at point 3, and for the present moment this is the
-approach taken. However, we can't be certain that we will get desirable performance characteristics
-from this alone, or that it will be as useful for client side applications as it might be for
-server-side applications (more on that in the next section).
+Presently, we have an invariant that all child modules of a parent module will complete evaluation before their
+parent module completes evaluation. What deferred evaluation would introduce is a new concept, where we would have a
```

Comment:
I'm not sure about this change. It suggests that this invariant has been changed. In my view it hasn't really. Child modules that are not lazy or that are shared are still evaluated eagerly. I would say it is closer to dynamic import in this regard.

Reply:
Thanks, I've updated this section and also reworked the background to draw further parallels between CommonJS and TLA.